src/fsharp/service/SemanticClassification.fs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.SourceCodeServices
+
+[<RequireQualifiedAccess>]
+type SemanticClassificationType =
+    | ReferenceType
+    | ValueType
+    | UnionCase
+    | Function
+    | Property
+    | MutableVar
+    | Module
+    | Printf
+    | ComputationExpression
+    | IntrinsicFunction
+    | Enumeration
+    | Interface
+    | TypeArgument
+    | Operator
+    | Disposable
+
+[<AutoOpen>]
+module TcResolutionsExtensions =
+    open System.Diagnostics
+    open System.Collections.Generic
+    open System.Collections.Immutable
+
+    open FSharp.Core.Printf
+    open FSharp.Compiler.AbstractIL.Internal.Library  
+
+    open FSharp.Compiler
+    open FSharp.Compiler.Range
+    open FSharp.Compiler.Tast
+    open FSharp.Compiler.Infos
+    open FSharp.Compiler.NameResolution
+    open FSharp.Compiler.ErrorLogger
+    open FSharp.Compiler.Lib
+    open FSharp.Compiler.PrettyNaming
+    open FSharp.Compiler.Tastops
+    open FSharp.Compiler.TcGlobals 
+    open FSharp.Compiler.SourceCodeServices.SymbolHelpers 
```

Comment:
Why are these nested within the module as opposed to being at the top of the file?

Reply:
I find it better to open specific namespaces/modules inside of modules like this as it doesn't open everything up for the whole file.